### PR TITLE
Add new content purpose sub and supergroups

### PR DESCRIPTION
**Pull request by fmlharrison**
As part of building the navigation topic pages we did work using the
JTBD framework to identify new content purpose sub and supergroups.
In order to help users find the content they require, we can group
document types by the needs they're designed to meet and the purpose of
the content.
See the wiki for more info:
https://gov-uk.atlassian.net/wiki/spaces/GFED/pages/305201156/Document+type+groupings

Due to the fact that the content purpose supergroups group subgroups not
document types, we have added a new yaml file for the supergroups.

The data lint specs have been change to test both data files.

### CODE REVIEW

File: data/content_purpose_supergroups.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Content Purpose Supergroups
+
+These are the grouping that the navigation team came up with
+as a result of an investigation done using the "Jobs to be Done" framework in Q4.
+
+There a 5 high level Content Purpose Supergroups with each group together some subgroups
```
Comment (vanitabarrett) [inline data/content_purpose_supergroups.md]: Typo: `a` should be `are`
Typo: `with` should be `which`

File: data/content_purpose_supergroups.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+More information about the JTBD framework and the supergroups can be found on the
+[GovUK wiki](https://gov-uk.atlassian.net/wiki/spaces/GFED/pages/305201156/Document+type+groupings)
+
+The list of which document types are grouped by both supergroups and subgroups can be
+found in the [supertypes.yml](./supertypes.yml) file.
+
+The list of the subgroups grouped by supergroup are in the file [supergroups.yml](./supergroups.yml).
+If you want to make a change to the `document_type` groupings in the sub or supergroups, you must make
+relevant change in both groups and both files to make sure there is no discrepancies between the groups.
```
Comment (vanitabarrett) [inline data/content_purpose_supergroups.md]: Typo: `relevant change` should be `the relevant change`
Typo: `there is` should be `there are`

File: data/supergroups.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+content_purpose_supergroup:
+  name: "Content Purpose Supergroup"
+  description: "These are the new super groups that came out of the research from the JTBD framework in Q4 by the navigation team and are used in the topic pages."
```
Comment (vanitabarrett) [inline data/supergroups.yml]: Need to remove the space between `super groups` to be `supergroups`

File: data/supertypes.yml [hunk 4]
```diff
@@ -381,3 +380,192 @@ government_document_supertype:
     - id: government-responses
       document_types:
         - government_response
+
+content_purpose_subgroup:
+  name: "Content Purpose Subgroup"
+  description: "These are the new sub groups that came out of the research from the JTBD framework in Q4 by the navigation team and are used in the topic pages. These are the sub groups connected to the content_purpose_document_supergroup"
```
Comment (vanitabarrett) [inline data/supertypes.yml]: Need to remove the space between sub groups to be subgroups

File: data/content_purpose_supergroups.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Content Purpose Supergroups
```
Comment (tijmenb) [inline data/content_purpose_supergroups.md]: Can these docs be added to the YAML file? That way they'll show up in the docs.

File: data/supertypes.yml [hunk 6]
```diff
@@ -381,3 +380,192 @@ government_document_supertype:
     - id: government-responses
       document_types:
         - government_response
+
+content_purpose_subgroup:
```
Comment (tijmenb) [inline data/supertypes.yml]: It might be worth marking `content_purpose_document_supertype` as deprecated so that there's less confusion while it's being removed.

File: data/content_purpose_supergroups.md (hunk 5 above)
Comment (fmlharrison) [reply data/content_purpose_supergroups.md]: Sure. I can do that.

File: data/content_purpose_supergroups.md (hunk 1 above)
Comment (fmlharrison) [reply data/content_purpose_supergroups.md]: Good spot.

File: data/content_purpose_supergroups.md (hunk 2 above)
Comment (fmlharrison) [reply data/content_purpose_supergroups.md]: Oh grammar

File: data/supergroups.yml (hunk 3 above)
Comment (fmlharrison) [reply data/supergroups.yml]: Cheers! 👀

File: data/supertypes.yml (hunk 6 above)
Comment (fmlharrison) [reply data/supertypes.yml]: Cool, I have marked this is deprecated.

File: data/supertypes.yml [hunk 7]
```diff
@@ -27,7 +27,7 @@ navigation_document_supertype:
 
 content_purpose_document_supertype:
   name: "Content purpose"
-  description: "As defined by the Frontend & Navigation team in Q3-2017. The team grouped document types that serve a similar purpose for users into top-level content groups. In time, this information could be used to serve relevant links into a content page sidebar and pull feeds of specific content groups into navigation pages."
+  description: "DEPRECIATED - Please use `content_purpose_subgroup`. The team grouped document types that serve a similar purpose for users into top-level content groups. In time, this information could be used to serve relevant links into a content page sidebar and pull feeds of specific content groups into navigation pages."
```
Comment (leenagupte) [inline data/supertypes.yml]: Typo. Should be `DEPRECATED` not `DEPRECIATED`. 👀

File: spec/data_lint_spec.rb [hunk 8]
```diff
@@ -47,4 +46,26 @@
       expect(document_types - allowed_document_types).to eql([])
     end
   end
+
+  describe "supergroups and subgroup document types" do
+    supergroups = YAML.load_file(File.dirname(__FILE__) + "/../data/supergroups.yml")
+    supertypes = GovukDocumentTypes::DATA
+
+    it "contains correct document types for both" do
```
Comment (leenagupte) [inline spec/data_lint_spec.rb]: I'd split this into three tests, so the test descriptions can be explicit about what it is that's being tested.

File: lib/govuk_document_types/version.rb [hunk 9]
```diff
@@ -1,3 +1,3 @@
 module GovukDocumentTypes
-  VERSION = "0.2.0".freeze
+  VERSION = "0.2.1".freeze
```
Comment (leenagupte) [inline lib/govuk_document_types/version.rb]: As @vanitabarrett mentioned elsewhere, this should be in a separate PR along with an update to the change note.

Also, I think this counts as minor change, rather than a patch, so should be updated to `0.3.0`.

File: spec/data_lint_spec.rb (hunk 8 above)
Comment (fmlharrison) [reply spec/data_lint_spec.rb]: I can't see how this would be split up. What would each be testing?

Comment (leenagupte) [reply spec/data_lint_spec.rb]: Had a conversation in meatspace with @fmlharrison about this.

**Comment by leenagupte** [comment]
Thanks for clearing things up @fmlharrison 👍
